app/models/manageiq/providers/openstack/inventory/collector/target_collection.rb

```diff
@@ -307,7 +314,26 @@ def infer_related_orchestration_stacks_ems_refs_db!
 
   def infer_related_orchestration_stacks_ems_refs_api!
     orchestration_stacks.each do |stack|
-      add_simple_target!(:orchestration_stacks, stack.parent, :tenant_id => stack.service.current_tenant["id"]) unless stack.parent.blank?
+      # Scan resources for VMs and add them as target, so the stack connects to vm, otherwise they don't connect on
+      # targeted refresh
+      orchestration_resources(stack).each do |resource|
+        case resource.resource_type
+        when "OS::Nova::Server"
+          # TODO(lsmola) we need to cache vms also on individual level, so the calling vms again has also these vms
+          add_simple_target!(:vms, resource.physical_resource_id)
```

Comment:
@mansam where do we use the `tenant_id` that I am seeing here. Cause I see it missing for other collections

Reply:
@aufi ^

Reply:
@Ladas It's used here: https://github.com/ManageIQ/manageiq-providers-openstack/blob/3209e0ffff8f60d9f311d58774f07827dd4cb0c1/app/models/manageiq/providers/openstack/cloud_manager/event_target_parser.rb#L68

If I remember correctly, with other collections, you can get the tenant from the event payload. I couldn't get the tenant from the event payload for stacks, so I did it this way,

Reply:
@mansam ok, can you check if you can add it? Also I've seen it being nil for some other targets, can you check that out with the OpenStack env I've send to @aufi ?

(feel free to just take this PR and continue :-))